go.mod

```diff
@@ -27,7 +27,6 @@ require (
 	k8s.io/code-generator v0.29.0
 	k8s.io/component-base v0.29.0
 	k8s.io/klog/v2 v2.110.1
-	sigs.k8s.io/controller-runtime v0.16.0
```

Comment:
Does removing controller-runtime still cause compile error on nsx-operator? https://github.com/vmware-tanzu/nsx-operator/issues/505

Reply:
> Does removing controller-runtime still cause compile error on dependency nsx-operator? 


Once this PR is merged, it will become indirectly dependent on `controller-runtime`, see here: https://github.com/kubernetes/cloud-provider-vsphere/blob/1071d14884564dae37eb7ab9ad1f99460052c90d/go.mod#L123  and NSX Operator API is only using `controller-runtime/pkg/scheme/scheme.go`, so as long as this file doesn't change we should be ok.

But for the long term, we still hope NSX Operator API can also get rid of `controller-runtime` as well

Reply:
Yup. In general API packages should have no dependencies on CR. 

In CAPI & CAPV we recently got rid of our CR imports in the API (xref: https://github.com/kubernetes-sigs/cluster-api/issues/9011)

Reply:
Looks like the client also has CR imports
```
 go mod graph  | grep sigs.k8s.io/controller-runtime
k8s.io/cloud-provider-vsphere sigs.k8s.io/controller-runtime@v0.14.5
github.com/vmware-tanzu/nsx-operator/pkg/apis@v0.0.0-20231227030711-34390278f3cd sigs.k8s.io/controller-runtime@v0.14.5
github.com/vmware-tanzu/nsx-operator/pkg/client@v0.0.0-20240103020117-417d1d01f14b sigs.k8s.io/controller-runtime@v0.14.5
```